tinygrad/renderer/cstyle.py

```diff
@@ -14,6 +14,7 @@ class CStyleLanguage(NamedTuple):
   smem_align: str = ""
   smem_prefix: str = ""
   smem_prefix_for_cast: bool = True
+  explicit_cast_alu: bool = False # hack for glsl
```

Comment:
What's special about GL that it needs this? Maybe the cast should always be called?

Reply:
I got rid of this hack, and solved it in `code_for_op` thanks to Quazalin's `dtype` refactor it was easy to solve it.
Got compiler errors for `bool * bool`, GLSL can't do it. So in `code_for_op` I do `int(a)*int(b)` (only for bool dtype). But then you'll have another problem, `cannot convert from 'int' to 'bool'` when you assign such result to a` bool alu`. That's why I added the explicit cast, but turns out I can just solve it in `code_for_op`.